### PR TITLE
Simpler MySQL inventory implementation.

### PR DESCRIPTION
**Pull request by appurist**
A new checked inventory provider specifically for MySQL (no Cassandra). Eliminates all Cassandra dependencies, simplifies code while continuing to use the "checked inventory" interface for security and reliability of inventories. Also removes some of the InWorldz-specific plugs that depended on Cassandra. Also implemented the `findUserTopLevelFolderFor` function for MySQL that was present in the Cassandra implementation but stubbed in MySQL.  Also updated `halcyon.sample.ini` with much simpler options now. For backwards-compatibility, if `inventory_source` is not found, it will use `legacy_inventory_source`, but documentation such as this sample should only mention `inventory_source` going forward.

Reasonably confident in this now, in spite of the large changes, since a build of this branch is the code in production on the Islandz grid, and has been running fine since grid startup.

### CODE REVIEW

**Comment by appurist** [comment]
Any idea why the Windows process in AppVeyor is getting an error? Looks like a failed folder move to bin? Maybe trying to overwrite bin instead of moving a file into it?

**Comment by kf6kjg** [comment]
It's actually trying to rename `bin` into `halcyon` for packaging purposes; renaming the `bin` folder into `halcyon` gives you a useful folder name when you decompress the ZIP.  However it looks like you created a folder named `Halcyon` to house the new MySQL lib, which makes sense.

**Comment by appurist** [comment]
Looks great good now!

File: Halcyon/Halcyon.Data.Inventory.MySQL/MysqlStorageImpl.cs [hunk 1]
```diff
@@ -255,7 +291,7 @@ public InventoryFolderBase getUserRootFolder(UUID user)
                         {
                             rootFolder = items[0];
                         }
-
```
Comment (kf6kjg) [inline Halcyon/Halcyon.Data.Inventory.MySQL/MysqlStorageImpl.cs]: Interesting.  Must have run across cases where the `Level` was wrong?

**Comment by kf6kjg** [comment]
All looks good to me too.  Merging.

File: Halcyon/Halcyon.Data.Inventory.MySQL/MysqlStorageImpl.cs (hunk 1 above)
Comment (appurist) [reply Halcyon/Halcyon.Data.Inventory.MySQL/MysqlStorageImpl.cs]: It just was never set. Level was something Tranq dreamed up I think, and initialized in Cassandra.  This code never set it, or left it at a default (can't quite remember, same thing either way). So you'll see a couple of new places where I initialize the `Level` to the known correct value. Line 80 does the same thing.

**Comment by kf6kjg** [comment]
Released (finally) in v0.9.42.

**Pull request merged**